src/main/java/com/google/sps/Recommend.java

```diff
@@ -20,10 +27,24 @@ private static void init() {
             .config("spark.master", "local[*]")
             .getOrCreate();
     spark.sparkContext().setLogLevel("ERROR");
+
+    datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   /** Rebuilds recommendation model and calculates recommendations for users. */
-  public static void calculateRecommend() {}
+  public static void calculateRecommend() {
+    if (spark == null || datastore == null) {
+      init();
+    }
+    Iterable<Entity> queriedUsers = datastore.prepare(new Query("User")).asIterable();
+    // get user prefs
+    Map<String, Map<String, Integer>> userPrefs = new HashMap<>();
+    for (Entity e : queriedUsers) {
```

Comment:
For debugging?

Reply:
That's just a method stub, I haven't gotten around to implementing anything yet.